tests/test_decorators.py

```diff
@@ -518,3 +518,33 @@ def validate_many(self, data, many):
         errors = schema.validate([{'foo': 3, 'bar': 'not an int'}], many=True)
         assert 'bar' in errors[0]
         assert '_schema' not in errors
+
+    def test_allow_reporting_field_errors_in_schema_validator(self):
+
+        class NestedSchema(Schema):
+            baz = fields.Int(required=True)
+
+        class MySchema(Schema):
+            foo = fields.Int(required=True)
+            bar = fields.Nested(NestedSchema, required=True)
+            bam = fields.Int(required=True)
+
+            @validates_schema(skip_on_field_errors=True)
+            def consistency_validation(self, data):
+                errors = {}
+                if data['bar']['baz'] != data['foo']:
+                    errors['bar'] = {'baz': 'Non-matching value'}
+
+                if data['bam'] > data['foo']:
+                    errors['bam'] = 'Value should be less than foo'
+
+                if errors:
+                    raise ValidationError(errors)
+
+        schema = MySchema()
+        errors = schema.validate({'foo': 2, 'bar': {'baz': 5}, 'bam': 6})
```

Comment:
I'm not confusing message parameter. What you're saying ("The ```errors``` value returned by ```load()``` is a named tuple") is incorrect. The _result_ of ```load()``` is named tuple out of which ```errors``` field is a dictionary of errors:
```python
class FooSchema(marshmallow.Schema):
    bar = marshmallow.fields.String(required=True)

FooSchema().load({'bar': 123})
# => UnmarshalResult(data={}, errors={'bar': [u'Not a valid string.']})
```
So UnmarshalResult is your named tuple while ```errors``` is a dictionary of errors.

https://marshmallow.readthedocs.io/en/dev/extending.html#schema-level-validation describes what will happen if you report a string error inside function decorated with ```validates_schema```. This stays the same with this PR. Storing errors on specific fields also works as before.

| The "inconsistency" between the two interfaces is fully documented. This PR is a breaking change.
Point me into documentation where it shows raising dicts in ValidationError and expecting it to not be treated as nested field errors.